src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyStringComparison.cs

```diff
@@ -81,6 +86,19 @@ public override void Initialize(AnalysisContext analysisContext)
                         return;
                     }
 
+                    if (linqExpressionType != null)
+                    {
+                        var enclosingLambdaOrLocalFunc = invocationExpression.GetAncestor(s_LambdaOrLocalFunctionKinds);
```

Comment:
@mavasani This new implementation should not cross lambdas boundaries.

Reply:
Thanks, this looks much simpler. Do you think it is worth moving the entire added code block here into an extension method on IOperation, say `IsInExpressionTree` or some such, given we might likely need more analyzers to bail out in expression tree context?

Reply:
Let's wait for the next use to create the helper.